pkg/recovery/recovery.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2022 IoTeX
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package recovery
+
+import (
+	"encoding/json"
+	"runtime/debug"
+	"time"
+
+	"github.com/shirou/gopsutil/load"
+	"github.com/shirou/gopsutil/v3/cpu"
+	"github.com/shirou/gopsutil/v3/disk"
+	"github.com/shirou/gopsutil/v3/mem"
+
+	"github.com/iotexproject/iotex-core/pkg/log"
+)
+
+type (
+	// CPUInfo stat cpu infos
+	CPUInfo struct {
+		PhysicalCount int             `json:"physical_count"`
+		LogicalCount  int             `json:"logical_count"`
+		TotalPercent  []float64       `json:"total_use_percent"`
+		PerPercent    []float64       `json:"per_use_percent"`
+		Loads         *load.AvgStat   `json:"average_loads"`
+		Times         []cpu.TimesStat `json:"running_times"`
+		Infos         []cpu.InfoStat  `json:"infos"`
+	}
+	// MemInfo stat memory infos
+	MemInfo struct {
+		Virtual *mem.VirtualMemoryStat `json:"virtaul"`
+		Swap    *mem.SwapMemoryStat    `json:"swap"`
+	}
+	// DiskInfo stat disk infos
+	DiskInfo struct {
+		IOCounters map[string]disk.IOCountersStat `json:"io_counters"`
+		Partitions []disk.PartitionStat           `json:"partitions"`
+	}
+)
+
+// Recovery get the stack infos of current goroutine and the cpu/mem/disk current infos
+func Recovery() {
+	if r := recover(); r != nil {
+		log.S().Errorf("recover error :%v", r)
+		log.S().Errorf("%s", string(debug.Stack()))
```

Comment:
the `debug.Stack()` will be the stack of the panic code?

Reply:
yes